includes/admin/class-amp-story-templates.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * AMP Story Templates handler.
+ *
+ * @package AMP
+ * @since 1.?
+ */
+
+/**
+ * Class AMP_Story_Templates
+ */
+class AMP_Story_Templates {
+
+	/**
+	 * Current version of the templates.
+	 *
+	 * @var string
+	 */
+	const STORY_TEMPLATES_VERSION = '0.1';
+
+	/**
+	 * Slug for templates' taxonomy.
+	 */
+	const TEMPLATES_TAXONOMY = 'amp_template';
+
+	/**
+	 * Slug for templates term.
+	 */
+	const TEMPLATES_TERM = 'story-template';
+
+	/**
+	 * Init.
+	 */
+	public function init() {
+		if ( ! post_type_exists( 'amp_story' ) ) {
+			return;
+		}
+
+		add_filter( 'rest_wp_block_query', array( $this, 'filter_rest_wp_block_query' ), 10, 2 );
+
+		$this->register_taxonomy();
+		$this->maybe_import_story_templates();
```

Comment:
@swissspidy, @westonruter What do you think of this kind of initial approach for the default templates?

Reply:
Sounds reasonable.

I wonder if we need some dynamic data in `get_story_templates()` as well at some point, e.g. default images, freshly generated UUIDs, etc. and how that works together with `template_exists`.

Reply:
Hmm, maybe. Do you think that using static UUIDs might cause issues?
On images: an option could be using static URLs and not Media Library.

Reply:
> Do you think that using static UUIDs might cause issues?

IDK. Perhaps someone exports their reusable blocks and someone imports them and then a conflict happens where two templates have the same IDs. Or maybe someone trashes a reusable block, somehow imports the templates again, and then ends up with the same situation.

> an option could be using static URLs and not Media Library.

I'd try to avoid the media library either way I think. Users wouldn't notice that an image belongs to a template and would most likely just delete it.

Static URLs means hosting them on `amp-wp.org` or a CDN, which might create some unnecessary overhead for us in maintaining that. Also, some people don't like loading assets from a CDN (e.g. emoji from WordPress.org, web fonts). Plus, it doesn't work offline.

An alternative would be bundling the images with the plugin and using `plugin_dir_url()` to reference them, making this a little bit more dynamic again.

Reply:
> An alternative would be bundling the images with the plugin and using plugin_dir_url() to reference them, making this a little bit more dynamic again.

That's what I actually had in mind by "static URLs", you're right that they're not really static.

> IDK. Perhaps someone exports their reusable blocks and someone imports them and then a conflict happens where two templates have the same IDs.

Hmm, maybe what's in the content of the post wouldn't matter since the block is cloned and assigned a new ID before inserting anyway?